src/calc_FD_CG_2DES.c

```diff
@@ -0,0 +1,385 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include <string.h>
+#include <time.h>
+#ifdef _WIN32
+    #include <Windows.h>
+#else
+    #include <unistd.h>
+#endif
+//#include "omp.h"
+#include "types.h"
+#include "NISE_subs.h"
+#include "polar.h"
+#include "calc_CG_2DES.h"
+#include "calc_FD_CG_2DES.h"
+#include <stdarg.h>
+#include "project.h"
+#include "propagate.h"
+#include "read_trajectory.h"
+#include "eq_den.h"
+//#include "mpi.h"
+#include "MPI_subs.h"
+#include <complex.h>
+
+void call_final_FD_CG_2DES(
+  t_non *non,float *P_DA,int segments,float *re_doorway,float *im_doorway,
+  float *re_window_SE, float *im_window_SE,float *re_window_GB, float *im_window_GB,
+  float *re_window_EA, float *im_window_EA,float *re_2DES, float *im_2DES, double *Q1, double *Q2){
+    /* Define variables for multiple waiting time use */
+    FILE *WTime;
+    char waittime[16];
+    int wfile;
+                            
+    /* Check if Waitingtime.dat is defined */
+    sprintf(waittime,"");
+    wfile=0;
+    if (access("Waitingtime.dat", F_OK) !=-1){
+      printf("Waitingtime file found by calc_FD_CG_2DES. \n");
+      WTime=fopen("Waitingtime.dat","r");
+      wfile=1;
+    }
+    /* While loop over waiting times */
+    
+    /* Change the value of non->tmax2 to the wanted waiting time */
+    while (wfile>-1){
+      /* Read new waiting time */
+      if (wfile==1){
+        if (fscanf(WTime,"%s",&waittime)==1){ // Should this not be waittime instead of &waittime?
+          printf("Calculating FD-CG2DES for %s fs \n",waittime);
+          non->tmax2 = floor(atof(waittime)/(non->deltat));
+        } else {
+          fclose(WTime);
+          break;
+        }
+      } else {
+        wfile=-1;
+      }
+      
+      CG_2DES_P_DA(non,P_DA,segments);
+      
+      FD_CG_full_2DES_segments(non,re_doorway,im_doorway,re_window_SE,im_window_SE,
+        re_window_GB,im_window_GB,re_window_EA,im_window_EA,P_DA,segments,waittime,wfile,Q1,Q2);
+    }
+}
+
+/* -=-._.-=-._.-=-._.-=-._.-=-._.-=-._.-=-._.-=-._.-=-. */
+/* READ IN QUANTUM YIELDS FROM SEPARATE Q1 AND Q2 FILES */
+/* -=-._.-=-._.-=-._.-=-._.-=-._.-=-._.-=-._.-=-._.-=-. */
+void read_in_QF_for_FD_CG_2DES(t_non *non, double *Q1, double *Q2, int segments){
```

Comment:
For consistency also define this function in the .h file.

void read_in_QF_for_FD_CG_2DES(t_non *non, double *Q1, double *Q2, int segments);